src/helpers/response-status.helper.ts

```diff
@@ -0,0 +1,36 @@
+import { HttpStatus } from '@nestjs/common';
+import { isAxiosError } from 'axios';
+
+/**
+ * Checks if the error object is an AxiosError and has a response status matching the specified status.
+ * @param {Object} options - An object containing the error and status.
+ * @param {Error} options.error - The error object to check.
+ * @param {HttpStatus} options.status - The status to compare against.
+ * @returns {boolean} - Returns true if the error is an AxiosError and has a matching response status, otherwise false.
+ */
+export const statusCheck = ({ error, status }: { error: Error; status: HttpStatus }): boolean => isAxiosError(error) && error.response?.status === status;
+
+/**
+ * Validates the error response against the specified search string.
+ * @param {Object} options - An object containing the error and search string.
+ * @param {Error} options.error - The error object to validate.
+ * @param {string} options.search - The search string to compare against the error message.
+ * @returns {boolean} - Returns true if the error response contains the search string, otherwise false.
+ */
+export const messageCheck = ({ error, search }: { error: Error; search: string }): boolean => {
+  if (!isAxiosError(error) || !error?.response?.data) {
```

Comment:
Im not sure we need the second boolean check as we check it in the if statements below anyway?

Reply:
This is useful to save any execution if the `response.data` does not exists.

Reply:
Cool -- Happy :)